packages/angular_devkit/build_angular/src/utils/index-file/inline-fonts.ts

```diff
@@ -0,0 +1,170 @@
+ */
+
+import * as cacache from 'cacache';
+import { readFile } from 'fs';
+import * as https from 'https';
+import { promisify } from 'util';
+import { findCachePath } from '../cache-path';
+import { cachingDisabled } from '../environment-options';
+import { htmlRewritingStream } from './html-rewriting-stream';
+
+const cacheFontsPath = cachingDisabled ? undefined : findCachePath('angular-build-fonts');
+const packageVersion = require('../../../package.json').version;
+
+const enum UserAgent {
+  Chrome = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36',
+  IE = 'Mozilla/5.0 (Windows NT 10.0; Trident/7.0; rv:11. 0) like Gecko',
+}
+
+const SUPPORTED_PROVIDERS = [
+  'https://fonts.googleapis.com/',
+];
+
+export interface InlineFontsOptions {
+  minifyInlinedCSS: boolean;
+  WOFFSupportNeeded: boolean;
+}
+
+export class InlineFontsProcessor {
+
+  constructor(private options: InlineFontsOptions) { }
+
+  async process(content: string): Promise<string> {
+    const hrefList: string[] = [];
+
+    // Collector link tags with href
+    const { rewriter: collectorStream } = await htmlRewritingStream(content);
+
+    collectorStream.on('startTag', tag => {
+      const { tagName, attrs } = tag;
+
+      if (tagName !== 'link') {
+        return;
+      }
+
+      // <link tag with rel="stylesheet" and a href.
+      const href = attrs.find(({ name, value }) => name === 'rel' && value === 'stylesheet')
+        && attrs.find(({ name }) => name === 'href')?.value;
+
+      if (href) {
+        hrefList.push(href);
+      }
+    });
+
+    await new Promise(resolve => collectorStream.on('finish', resolve));
+
+    // Download stylesheets
+    const hrefsContent = await this.processHrefs(hrefList);
+    if (hrefsContent.size === 0) {
+      return content;
+    }
+
+    // Replace link with style tag.
+    const { rewriter, transformedContent } = await htmlRewritingStream(content);
+    rewriter.on('startTag', tag => {
+      const { tagName, attrs } = tag;
+
+      if (tagName !== 'link') {
+        rewriter.emitStartTag(tag);
+
+        return;
+      }
+
+      const hrefAttr = attrs.some(({ name, value }) => name === 'rel' && value === 'stylesheet')
+        && attrs.find(({ name, value }) => name === 'href' && hrefsContent.has(value));
+      if (hrefAttr) {
+        const href = hrefAttr.value;
+        const cssContent = hrefsContent.get(href);
+        rewriter.emitRaw(`<style type="text/css" title="${href}">${cssContent}</style>`);
+      } else {
+        rewriter.emitStartTag(tag);
+      }
+    });
+
+    return transformedContent;
+  }
+
+  private async getResponse(url: string, userAgent: UserAgent): Promise<string> {
+    const key = `${packageVersion}|${url}|${userAgent}`;
+
+    if (cacheFontsPath) {
+      const entry = await cacache.get.info(cacheFontsPath, key);
+      if (entry) {
+        return promisify(readFile)(entry.path, 'utf8');
```

Comment:
consider calling `promisify(readFIle)` once at the top of the file and save as a variable